packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -150,6 +150,29 @@ class TextEditingController extends ValueNotifier<TextEditingValue> {
     );
   }
 
+  /// Builds [TextSpan] from current editing value.
+  ///
+  /// By default makes text in composing range appear as underlined.
+  /// Descendants can override this method to customize appearance of text.
+  TextSpan buildTextSpan({TextStyle style , bool withComposing}) {
+    if (value.composing.isValid && withComposing) {
+      final TextStyle composingStyle = style.merge(
+        const TextStyle(decoration: TextDecoration.underline),
+      );
+      return TextSpan(
+        style: style,
+        children: <TextSpan>[
+          TextSpan(text: value.composing.textBefore(value.text)),
+          TextSpan(
+            style: composingStyle,
+            text: value.composing.textInside(value.text),
+          ),
+          TextSpan(text: value.composing.textAfter(value.text)),
+      ]);
+    }
+    return TextSpan(style: style, text: text);
```

Comment:
I thought that the style guide had something about using an early return in cases like this, but I couldn't find it so I'm unsure.  Up to you.

```dart
if (!value.composing.isValid || !withComposing) {
  return TextSpan(style: style, text: text);
}
final TextStyle...
...
```